tools/exec_trace_parser.py

```diff
@@ -1,61 +1,133 @@
 class TraceReaderInterface:
+    """Interface expected by ExecTraceParser for returning trace buffer values.
+
+    Users of this module must extend this class to provide trace buffer values
+    to ExecTraceParser. How values are retrieved from an execution tracer
+    backend is implementation specific.
+    """
+
+    """Value to return after the last value has been read."""
+    END_OF_TRACE_BUFFER = -1
+
     def read_next(self) -> int:
         """Return the next value from the trace buffer.
 
-        The value must be returned as an integer regardless
-        of its intermediate format. This function may block
-        if waiting for new values.
+        This function is allowed to block indefinitely. This is to allow for
+        use in live trace scenarios where the trace reader will be continually
+        waiting for new values to arrive.
+
+        Returns:
+          The next value from the trace buffer, or END_OF_TRACE_BUFFER.
         """
         pass
 
 class ExecTraceParser:
+    """Translates trace buffer values into human readable text."""
     def __init__(self, functions, variables, registers):
+        """Initialize the parser with all of the look-up dictionaries.
+
+        Args:
+          functions: Dictionary that maps MCU addresses to function names.
+          variables: Dictionary that maps MCU addresses to variable names.
+          registers: Dictionary that maps MCU addresses to
+                     parse_svd.PeripheralRegister objects.
+          """
         self.functions = functions
         self.variables = variables
         self.registers = registers
+        # The parser needs to know the MCU's base addresses for flash, RAM and
```

Comment:
I would add this information to the docstring.

Reply:
Updated.